datadotworld/client/api.py

```diff
@@ -733,10 +731,14 @@ def append_records(self, dataset_key, stream_id, body):
         >>> api_client.append_records('username/test-dataset','streamId', \
         >>> {'content':'content'})
         """
+        streams_api_client = self._content_negotiating_client
+        streams_api_client.default_mimetype = desired_mimetype
```

Comment:
This is not correct.
There is an important difference between content negotiation in the `sql()` and `sparql()` case and this.
Here, we want to allow users to change the `Content-Type` header of the request as opposed to the `Accept` header. In other words, our `/streams` endpoint accept payloads that can be `application/json` or `application/json-l`.

Reply:
Yes, that is what happens in here. If a user specifies a desired_mimetype, the content-type is updated and not accept.